specification/trace/semantic_conventions/http.md

```diff
@@ -153,6 +153,16 @@ This span type represents an outbound HTTP request. There are two ways this can
 
 For an HTTP client span, `SpanKind` MUST be `Client`.
 
+> **Note**
+> It is nice for HTTP client spans to represent the complete lifetime of the HTTP request.
+> An HTTP request ends once the response body has been fully read or the response closes with an error.
+> However, if there is any possibility for application code to not fully read the HTTP response
+> (and for the HTTP client library to then have to clean up the HTTP response asynchronously),
+> the HTTP client span SHOULD be ended earlier (e.g. after the HTTP response headers are read, or fail to be read).
+> This avoids having to end the span asynchronously later on at a time which is no longer directly associated with the application code which made the HTTP request.
+>
+> Because of the potential for confusion around this, HTTP client library instrumentations SHOULD document their behavior around ending HTTP client spans.
```

Comment:
I'm OK with this, however I want to encourage us to think about longer term (not blocking this PR though):
1. I think the HTTP client library instrumentations SHOULD document the exact behavior/expectation, not just limited to "the behavior around ending HTTP client spans". For example, how to measure start is also interesting, and there could be more interesting things (e.g. if there is HTTP.SYS, TDI, IIS and ASP.NET involved, what exactly is the start/end time and how errors are handled). This is a great example of what I would love to see https://www.w3.org/TR/navigation-timing/.
2. I think we should encourage all instrumentations to document what does "duration" mean, this is not specific to HTTP (I guess we're seeing more issues with HTTP right now simply because HTTP is the first semantic convention we're trying to stabilize).

Reply:
For 1: An HTTP request is quite a high-level concept. Yes, it should include DNS, opening a TCP connection, TLS handshake, etc.

Reply:
> For example, how to measure start is also interesting (is DNS resolution included, do we consider TCP connection, TLS handshake, etc.),

I believe this depends on how the instrumentation is written; see the [two types of HTTP client instrumentations](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/trace/semantic_conventions/http.md#http-client).

For the low-level HPP instrumentations, I'd say that the HTTP span MUST NOT include connection/DNS/TLS because they all happen before the request is actually sent over the wire; and if we wish to include these details then we should have separate spans for each of these low-level operations (see https://github.com/open-telemetry/opentelemetry-specification/issues/3155).

Reply:
👍 I've generalized this PR to HTTP client duration and included a note about start time as well.